wp-dependency-installer.php

```diff
@@ -83,30 +110,37 @@ public function load_hooks() {
 		/**
 		 * Register wp-dependencies.json
 		 *
-		 * @param string $plugin_path Path to plugin or theme calling the framework.
+		 * @param string $plugin_path (optional) Path to plugin or theme calling the framework.
 		 */
-		public function run( $plugin_path ) {
+		public function run( $plugin_path = false ) {
+			if ( ! $plugin_path ) { // TODO: deprecate parameter in future releases?
```

Comment:
How would you plan on deprecating this? Is it simply a matter of calling the class differently?

Reply:
```php
// Next major release.
WP_Dependency_Installer::instance( __DIR__ )->register( $config );
WP_Dependency_Installer::instance( __DIR__ )->run();
```

Reply:
> How would you plan on deprecating this? Is it simply a matter of calling the class differently?

I think that, in the long term, this way of calling can be considered more flexible and clearer to understand:

```php
// Associative array
WP_Dependency_Installer::instance( __DIR__ )->register( $config );
WP_Dependency_Installer::instance( __DIR__ )->run();

// wp-dependecies.json
WP_Dependency_Installer::instance( __DIR__ )->run();
```

I don't think it can be deprecated at once just because I noticed that errors can be generated if you have multiple plugins that use / include different versions of this framework ( although this is still a general problem ).

Also, for example, it is not possible to "easily determine" which of the two versions it is to run, ( but that is always executed whose including plugin has alphabetic precedence )

Reply:
however yes, I would keep it in the code but I would recommend using the new version